src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let separatedString = '';
+    const lastIndex = this.length - 1;
+
+    for (let i = 0; i < lastIndex; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        separatedString += `${separator}`;
+      } else {
+        separatedString += `${this[i]}${separator}`;
+      }
```

Comment:
split this if in two:
- one to check if you need to add the element, do it ONLY if it's NOT null or undefined
- second one to check if you need to add the separator, do it ONLY if you're NOT working with the LAST element